### PR TITLE
Use simpler tab trap in date picker

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

The simpler tab trap implementation reduces implicit coupling between date picker dropdown and calendar elements. The change is to support the introduction of the embedded date picker variant.

### How has this been tested?

Existing tests

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

[*Attach any related links/pull request for this change*]

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/).

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts).

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/internal/components/tab-trap/index.tsx [hunk 1]
```diff
@@ -18,4 +18,44 @@ const TabTrap = ({ focusNextCallback, disabled = false }: TabTrapProps) => {
...
+  return focusableElements[0];
+}
+
+function getLastFocusableElement(container: HTMLElement) {
+  const focusableElements = getFocusableElements(container);
+  return focusableElements[focusableElements.length - 1];
+}
+
+function getFocusableElements(container: HTMLElement) {
+  return container.querySelectorAll('button, input, select, textarea, [href], [tabindex]:not([tabindex="-1"])');
```
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/internal/components/tab-trap/index.tsx:58]: That looks great, I am happy to adopt it as part of this PR 👍

[the remark this answers is not in the document]
Comment (pan-kot) [reply src/internal/components/tab-trap/index.tsx]: So I didn't fully understand the focus-lock implementation and instead I've taken a simplified version.

I think it would make sense to add the complete one separately with test coverage on those corner cases that you mentioned. We can also consider adding it to the toolkit, wdyt?

File: src/internal/components/tab-trap/focus-lock.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (just-boris) [inline src/internal/components/tab-trap/focus-lock.tsx:1]: Question about placement of this file, why is it `components/tab-trap/focus-lock` and not simpler `components/focus-lock`

File: src/internal/components/tab-trap/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (just-boris) [reply src/internal/components/tab-trap/index.tsx]: It is not very great to have semi-working implementation (because if not now, where else do we have time to finish this), but I believe you know what you are doing with this functionality. Approved

**Review by just-boris — APPROVED** [review]
Approved with follow-up comments

File: src/internal/components/tab-trap/focus-lock.tsx (hunk 2 above)
Comment (pan-kot) [reply src/internal/components/tab-trap/focus-lock.tsx:1]: Agree -- will update 👍

File: src/internal/components/tab-trap/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pan-kot) [reply src/internal/components/tab-trap/index.tsx]: There are more places that can benefit from the new focus-lock approach. I will create a BRF task to migrate those to the new implementation with adding the necessary test coverage.

**Comment by pan-kot** [comment]
Rebased with main

**Review by just-boris — APPROVED** [review]
